\documentclass[aps,rmp, twocolumn]{revtex4}
%\documentclass[a4paper,10pt]{scrartcl}
%\documentclass[aps,rmp,twocolumn]{revtex4}

\usepackage[utf8]{inputenc}
\usepackage{amsmath,graphicx}
\usepackage{color}
\usepackage{booktabs}
%\usepackage{cite}

\newcommand{\bq}{\begin{equation}}
\newcommand{\eq}{\end{equation}}
\newcommand{\bn}{\begin{eqnarray}}
\newcommand{\en}{\end{eqnarray}}
\newcommand{\Richard}[1]{{\color{red}Richard: #1}}
\newcommand{\gene}[1]{{\it #1}}
\newcommand{\mat}[1]{{\bf #1}}
\newcommand{\vecb}[1]{{\bf #1}}
\newcommand{\abet}{\mathcal{A}}
\newcommand{\eqp}{p}
\newcommand{\LH}{\mathcal{L}}


\begin{document}
\title{Contributions of adaptation and purifying selection to SARS-CoV-2 evolution}
\author{Richard A.~Neher}
\address{Biozentrum, University of Basel, Basel, Switzerland \looseness=-5}
\address{Swiss Institute of Bioinformatics, Switzerland \looseness=-5}

\date{\today}

\begin{abstract}
Continued evolution and adaptation of SARS-CoV-2 has lead to more transmissible and immune-evasive variants with profound impact on the course of the pandemic.
Here I analyze the evolution of the virus over 2.5 years since its emergence and estimate rates of evolution for synonymous and non-synonymous changes separately for evolution within clades -- well defined mono-phyletic groups with gradual evolution -- and for the pandemic overall.
The rate of synonymous mutations is found to be around 6 changes per year.
Synonymous rates within variants vary little from variant to variant and are compatible with the overall rate of 7 changes per year (or $7.5\times 10^{-4}$ per year and codon).
In contrast, the rate at which variants accumulate amino acid changes (non-synonymous mutation) was initially around 12-16 changes per year, but in 2021 and 2022 dropped to 6-9 changes per year.
The overall rate of non-synonymous evolution, that is across variants, is estimated to be about 26 amino acid changes per year (or $2.7\times 10^{-3}$ per year and codon).
This strong acceleration of the overall rate compared to within clade evolution indicates that the evolutionary process that gave rise to the different variants is qualitatively different from that in typical transmission chains and likely dominated by adaptive evolution.
I further quantify the spectrum of mutations and purifying selection in different SARS-CoV-2 proteins and show that the massive global sampling of SARS-CoV-2 is sufficient to estimate site specific fitness costs across the entire genome.
Many accessory proteins evolve under limited evolutionary constraint with little short term purifying selection.
About half of the mutations in other proteins are strongly deleterious.
\end{abstract}

\maketitle

Since its emergence in late 2019 \citep{zhu_novel_2020}, SARS-CoV-2 has displayed a discontinuous pattern of evolution with large jumps in sequence space giving rise to phylogenetically distinct variants \citep{hodcroft_spread_2021,volz_assessing_2021,tegally_detection_2021,faria_genomics_2021,naveca_covid-19_2021,viana_rapid_2022}.
Many of these variants spread considerably faster and quickly displaced the resident variants at the time either because of intrinsically increased transmissibility, evasion of prior immunity in the population, or a combination of both.
Specific Variants of Concern (VoC) or Interest (VoI) were designated by the WHO and labeled by Greek letters \citep{konings_sars-cov-2_2021}.
The branches leading to these variants are characterized by many amino acid-changing mutations that often cluster in the S1 domain of the spike protein \citep{kistler_rapid_2022}.

This pattern of rapid non-synonymous evolution in viral surface proteins that interact with the host cells is common among many RNA viruses and for example well studied in influenza A virus evolution \citep{bhatt_genomic_2011,strelkowa_clonal_2012}.
But adaptive evolution of influenza viruses tends to be gradual without large jumps in sequence space, while new variants of SARS-CoV-2 with tens of novel mutations emerged suddenly without intermediate genomes being observed, the most dramatic being the emergence of Omicron in late 2021 \citep{viana_rapid_2022}.
The pre-dominant hypothesis for this cryptic emergence of highly mutated, transmissible and immune evasive variants are chronic infections that are common in patients with impaired immune systems, either through HIV-1 infection \citep{cele_sars-cov-2_2022} or medical intervention \citep{choi_persistence_2020,kemp_sars-cov-2_2021}.
During such chronic infections, extensive intra-host diversity can develop through accelerated evolution
\citep{chaguza_accelerated_2022}.
Onward transmission from such chronic infections has also been documented \citep{gonzalez-reiche_intrahost_2022}.
However, to date there is no direct evidence for the mode of emergence of any variant.
The case for chronic infection being an important contributor is strongest for the variants Alpha and Omicron \citep{hill_origins_2022}.

The dichotomous pattern of SARS-CoV-2 evolution with step-wise evolution within variants and atypical bursts of evolution leading to new variants has been investigated by \citet{tay_emergence_2022} and \citet{hill_origins_2022}, who showed that the rate of evolution along branches giving rise to new variants is up to four-fold higher than the background rate.
Here, I build on these results and investigate the patterns of SARS-CoV-2 diversification within variants and compare these to the global dynamics of evolution and adaptation.
This comparison reveals a consistent dichotomy between slow within-variant evolution and rapid adaptive evolution giving rise to new variants.
This difference in evolutionary rate is only seen for non-synonymous changes -- the rate of synonymous evolution within variants is compatible with that seen between variants.
Furthermore, early variants display more rapid non-synonymous evolution than later variants suggesting more ubiquitous adaptive evolution early on.
I further quantify the level of functional constraint of different open reading frames and infer a map of mutational tolerance across the genome from patterns of rare diversity.

\begin{figure}
    \includegraphics[width=0.5\textwidth]{figures/clade_overview.png}
    \caption{{\bf Schematic summary of Nextstrain clades.}\label{fig:clade_overview}}
\end{figure}

\begin{figure*}[tb]
    \includegraphics[width=\textwidth]{figures/rtt/20C_rtt.pdf}
    \includegraphics[width=\textwidth]{figures/counts/20C_counts.pdf}
    \caption{{\bf Within-clade divergence increases linearly with time (example clade 20C).} Top: Each panel shows the number of within-clade mutations (total (A), amino acid changing (B), synonymous (C)) as a function of time.
    The green line in panel A indicates the divergence cut-off, panels B\&C only show sequences that pass the divergence filter. Each panel also shows mean $\pm$ standard deviation and a weighted linear fit.
    Bottom: Panels D and E show the prevalence of the most common genotypes (D) and mutations (E) during the first 3 months. Genotypes in (D) refer to sequences with {\bf all and only} the mutations indicated, while mutations in (E) count all sequences with a specific mutation, regardless of mutations elsewhere.
    In the case shown (20C), the founder genotype initially dominates and no specific mutation ever reaches high frequency. Note that specific genotypes invariably decrease in frequency due to subsequent mutations, while mutations irrespective of their genetic background persist. Panel F shows a Poisson model fit to the breakdown of the population into genotypes with different number of mutations over time. Analogous figures for other clades are included in the appendix.
    \label{fig:within_clade}}
\end{figure*}

\section*{Results}

Evolutionary rates and divergence times are typically estimated using phylogenetic approaches \citep{drummond_relaxed_2006}.
These methods, however, cannot handle the volume of SARS-CoV-2 data available and data have to be dramatically down-sampled.
Furthermore, phylogenetic methods impose an hierarchical structure on the data and are thus very sensitive to problematic sequences or metadata: Any mis-dated, recombinant, contaminated, or otherwise chimeric sequence can fatally distort the analysis.
Problematic sequences can be particularly common when a new variant takes over since sequencing protocols need adjusting \citep{deMaio_issues_2020}.

To circumvent many of the above-mentioned problems and still use the majority of the available data, I use a combination of automated filtering and simple robust approaches to analyze the evolutionary patterns (see Materials and Methods).
I first use Nextclade \citep{aksamentov_nextclade_2021} to assign 12 million sequences available in GISAID (2022-07-25) \citep{shu_gisaid_2017} to one of the Nextstrain clades \citep{hadfield_nextstrain_2018,roemer_sars-cov-2_2022}, which are denoted by a Year-Letter combination (e.g.~19A, 20A, 20B, \ldots), see Fig.~\ref{fig:clade_overview}.
Sequences belonging to recombinant Pango lineages are excluded \citep{rambaut_dynamic_2020}.
These clades represent well-defined groups of sequences with little evidence of recombination within them and are analyzed independently.
In this analysis, I only consider clades that have had significant circulation for at least 6 months.

For each clade, I define a ``founder'' genotype and exclude any sequence that does not have the full set of clade defining mutations relative to the reference sequence Wuhan/Hu-1.
This founder sequence is manually curated for each clade considered.
This filtering removes most incomplete sequences as well as sequences where amplicon drop-out are back-filled with the reference sequence, but will ignore a few sequences with true reversion mutations.
In addition, sequences with a Nextclade QC score above 30 (80 for 21H because of an unaccounted frameshift) are removed.
Results are insensitive to the stringency of this filtering.
For this reduced set of sequences, I determine the mutations they carry on top of the founder genotype of the clade to analyze diversification and divergence within the clade.
The latter step is done for nucleotide changes as well as for amino acid changes.

Within each clade, the number of mutations is expected to increase linearly in time and the variation around this mean would, in an ideal case, obey Poisson statistics.
For the majority of sequences, this is approximately true, but some problematic sequences have more mutations than expected.
To exclude these outliers, I perform a simple linear regression of the number of ``intra-clade'' mutations against time and remove sequences whose deviation from the linear fit exceeds twice the expected standard deviation by 3 mutations (see Fig.~\ref{fig:within_clade}). These outliers are never more than 1\% of all sequences that pass the first round of filtering and the results are insensitive to the filtering criteria.


After removing these outliers, the data are binned by calendar week.
Evolutionary rate and putative emergence date of the variant are then estimated by weighted linear regression where each time bin is weighted with the fourth root of the number of sequences in the bins.
The exact functional form of this weighting does not have a big influence on the results, but a sublinear weighting helps to counter the large variation in sequencing effort across countries and the unavoidable imbalance due to the fact that few sequences are available early on when an emerging variant is still rare.
Fig.~\ref{fig:within_clade} shows the increasing intra-clade divergence for clade 20C, a large clade that emerged in early 2020 that was common in North America and Scandinavia.
Both synonymous and non-synonymous within-clade average divergence increase linearly over time allowing for a robust estimate of the rate.

Due to shared ancestry, divergences of sequences are not independent data points and a regression against time is general not a suitable method to estimate evolutionary rates.
In particular, confidence intervals are difficult to obtain.
However, in the case of rapidly expanding variants we typically observe a large number of independent lineages emanating from one or several basal polytomies.
Along each of these lineages, mutation accumulation is independent.
While not every sequence is an independent sample, the effective number of independent samples is large and the steadily increasing average divergence allows to estimate the rate robustly.

A simple model for diversity within a growing variant is a super-critical branching process with growth rate $\alpha$ and an embedded mutation process.
Offspring of genomes with $i$ mutations will carry $i+j$ mutations, where $j$ is a Poisson distributed number with mean $\mu t$ (mutation rate $\mu$ and generation time $t$).
The probability that offspring genomes are different from their parents is $u = 1-e^{-\mu t}$, which for a generation time of $t=5$ days and a rate of $\mu = 15/year$ evaluates to $u\approx 0.2$ (note that this rate excludes strongly deleterious mutations, see Discussion).

When considering a rapidly growing well sampled outbreak, typically a single founder genotype will give rise to a large number of daughter lineages that evolve independently.
In this case, the diversification processes is robustly described by its mean.
Since the above branching process is linear, the mean number of cases $n$ will increase exponentially with rate $\alpha$, while the number of genomes with $i$ mutations relative to the founder $m_i$ grows with rate $\alpha - u$ per generation:
\begin{equation}
    \begin{split}
        \frac{dn}{dt} &= \alpha n \\
        \frac{dm_0}{dt} &= (\alpha - u) m_0 \\
        \frac{dm_1}{dt} &= u m_0 + (\alpha - u) m_1 \\
        \frac{dm_2}{dt} &= u m_1 + (\alpha - u) m_2 \\
        \cdots
    \end{split}
\end{equation}
with solution $m_i = e^{(\alpha - u)t} \frac{(ut)^i}{i!}$.
Note that this model assumes continuous time and only allows increments by one mutation at a time.
Multiple mutations in one serial interval can still happen through successive mutations within one host.
At time $t$ after the emergence of the variant, the number of mutations in the population is expected to be Poisson distributed with mean $ut$.
The overall number of cases is $e^{\alpha t}$ or more generally $e^{\int \alpha(t) dt}$ if growth rate or ascertainment varies over time.

For some clades, especially those that are well sampled soon after their emergence, this Poisson model is a good fit to diversity accumulation and yields estimates of rates and time of origin that are compatible with the divergence regression, see Fig.~\ref{fig:within_clade}F for clade 20C.
In this case, the founder genotype initially dominates, but is gradually replaced -- first by single mutant genotypes, then double mutants, and so forth (comp.~Fig.~\ref{fig:within_clade}).
Analogous graphs for all other clades considered are included in the appendix.

In other cases, notable subclades did become dominant during the early stochastic dynamic.
In this case, extrapolation of the linear fit to zero divergence is not necessarily a good estimator of the emergence time of a variant: if branches leading to big subclades carry anomalously many or few mutations, the divergence time will be over- or underestimated.
For several clades 19B, 20H (Beta), 21D (Eta), 21G (Lambda), 21H (Mu), 21I (Delta), 21J (Delta), 21L (Omicron, BA.2), and 22B (Omicron, BA.5), founder-like variants are a minority even in early data.

These Poisson weights of mutation numbers are again only valid if mutations accumulate along many independent lineages.
In particular, this assumption is violated if some lineages spread systematically faster than others either because of epidemiological factors or because they carry adaptive mutations.
In variant 21K (Omicron, BA.1), a sublineage with mutation \texttt{S:R346K} might have enjoyed a transmission advantage.
A possible consequence of this advantage is visible in Supp.~Fig.~\ref{fig:21K_divergence} where genotypes with 0, 1, or 2 mutations are decaying more rapidly than those with more mutations.

\begin{figure*}
    \includegraphics[width=0.9\textwidth]{figures/rate_summary.pdf}
    \caption[]{{\bf Divergence and evolutionary rates of different Nextstrain clades.} Panels A,B\&C show the estimated divergence of the founder genotype of each clade (big dot or square) and the subsequent divergence trend for all nucleotide changes, amino acid changes, and synonymous changes, respectively.
    In addition, each panel contains a regression of the divergence of clade founders vs time (gray line).
    The standard deviation expected based on Poisson statistics is indicated as shaded area.
    Panel D summarizes the individual rate estimates (dots and squares) and compares them to the estimate of inter-clade rates (gray lines).
    In panel D, clades are in alphabetical order, which is similar to their (uncertain) order of emergence.
    The red outlier is `composite' clade 19B++ (containing 19B, 19A, 20A, 20B, 20C) with inflated rates due to adaptive mutations on the branch leading to clade 20A.
    \label{fig:rate_summary} }
\end{figure*}

Despite these caveats, for almost all Nextstrain clades, the slope at which diversity increases is robust (similar patterns as Fig.~\ref{fig:within_clade}A-C for clade 20C), allowing us to estimate clade-specific evolutionary rates for amino acid and synonymous changes.
These rates are summarized in Fig.~\ref{fig:rate_summary} and Tab.~\ref{tab:rates}.
In addition to these within-clade rates, I estimated the rate at which the clades themselves accumulated amino acid and synonymous changes by regressing the number of differences of the clade's founder sequence (relative to the putative root in clade 19B \citep{caraballo-ortiz_tophap_2022}) against the estimated time of origin of the clade.
These regressions are shown in thick gray lines in Fig.~\ref{fig:rate_summary}.

Rates of synonymous change are very consistent across clades (about 5-8 changes per genome per year) and also agree with the overall rate of synonymous changes of 7.3 changes per genome per year.
The rates of non-synonymous changes are much more variable (Fig.~\ref{fig:rate_summary}D).
Within clades, the rate of non-synonymous changes varies between 5 and 16 changes per year.
Earlier clades are estimated to have larger rates around 10-15 changes per year, while rate estimates for later clade fall  between 3 to 9 changes per year (see Fig.~\ref{fig:rate_progression}).
In contrast, the inter-clade non-synonymous rate exceeds 25 changes per year.
The spike protein does not contribute to this decreasing trend in the rate of within-clade non-synonymous evolution, which is most evident in \emph{ORF1ab} and to a lesser degree in accessory proteins and \emph{N} (see Supp.Fig.~\ref{fig:rate_progression_by_gene}).

Nextstrain clades tend to be defined by long branches leading to a large polytomy.
It could thus be that the estimated inter-clade rate exceeds the intra-clade rate purely because of this conditioning.
This effect might be particularly important early on in the pandemic when diversity was low and when branches with as few as two mutations were used to define new clades.
I therefore also included composite clades 19B+, 19B++ and 20A+ containing sequences from 19A and 19B (rooted on 19B), 19A, 19B, 20A, 20B, 20C, 20D (rooted on 19B), and 20A, 20B, 20C, 20D (rooted on 20A).
The estimates for composite clades 19B+ and 20A+ are consistent with the estimates of the individual clades, while apparent rates of 19B++ are considerably higher.
The latter is due to the rapid expansion and subsequent dominance of clade 20A, the first clade with spike mutation $D614G$, and its descendants which rapidly fixed four additional mutations \citep{korber_tracking_2020}.
This is an early example of an accelerated global rate of evolution due to adaptive evolution, see Discussion.

At the other extreme, some Omicron clades have very low rate estimates which should be interpreted carefully.
Founder genotypes of Omicron clades 21L (BA.2), 22A (BA.4) and 22B (BA.5) are rarely sampled.
These clades showed considerable diversity, including reversions to the reference, soon after their initial discovery \citep{tegally_emergence_2022}.
This diversity and the complex, possibly recombinant, origin of the Omicron clades makes estimating their rates of evolution challenging.


\begin{table*}
\begin{tabular}{l|rrrrrrr}
    \hline
    {\bf clade} & \# of seq. & overall rate $[y^{-1}]$ & aa rate $[y^{-1}]$ &  syn rate $[y^{-1}]$ & overall div. &  aa div. &  syn div. \\
    \hline
    19B &           8187 & 19.46 &    12.37 &      7.09 &        0 &       0 &        0 \\
   19B+ &           22745 & 18.56 &    11.92 &      6.64 &        0 &       0 &        0 \\
  19B++ &           126823 & 27.91 &    16.41 &     11.50 &        0 &       0 &        0 \\
  19A &             14382 & 19.73 &    12.79 &      6.95 &        2 &       1 &        1 \\
  20A &             44629 & 15.08 &     7.94 &      7.15 &        6 &       3 &        3 \\
    20B &           42077 & 14.29 &     7.69 &      6.59 &        9 &       5 &        4 \\
    20C &           47865 & 16.73 &     9.92 &      6.81 &        8 &       5 &        3 \\
   20A+ &           136863 & 16.30 &     9.40 &      6.90 &        6 &       3 &        3 \\
    20E &           82240 & 14.04 &     5.65 &      8.39 &       13 &       5 &        8 \\
    20H (Beta) &    5077 & 15.48 &     9.51 &      5.96 &       21 &      16 &        5 \\
    20I (Alpha) &   277226 & 11.56 &     5.49 &      6.07 &       30 &      19 &       11 \\
    20J (Gamma) &   24970 & 12.97 &     6.29 &      6.68 &       35 &      22 &       13 \\
    21D (Eta) &     3653 & 11.36 &     6.29 &      5.07 &       27 &      13 &       14 \\
    21G (Lambda) &  4567 & 12.04 &     6.51 &      5.53 &       28 &      19 &        9 \\
    21H (Mu) &      2897 & 18.25 &    10.05 &      8.20 &       31 &      21 &       10 \\
    21I (Delta) &   86585 & 16.55 &     8.93 &      7.62 &       30 &      24 &        6 \\
    21J (Delta) &   733825 & 14.30 &     7.17 &      7.14 &       34 &      28 &        6 \\
    21K (Omicron) & 956696 & 13.48 &     6.45 &      7.03 &       56 &      46 &       10 \\
    21L (Omicron) & 690177 & 9.24 &     4.39 &      4.85 &       68 &      50 &       18 \\
    22A (Omicron) & 33451 & 7.71 &     2.81 &      4.90 &       71 &      52 &       19 \\
    22B (Omicron) & 124239 & 11.90 &     5.03 &      6.87 &       66 &      50 &       16 \\
           \hline
    \end{tabular}
\caption{{\bf Evolutionary rates estimates from root-to-tip regressions for overall nucleotide changes, amino acid changes, and synonymous changes.}
The column `\# of seq.' shows the number of sequences that entered the analysis after filtering and restriction to the first 6 months after the emergence of the variant.
    The last three columns give the distances of the clade founder sequence from putative MRCA of SARS-CoV-2 (19B).
\label{tab:rates}}
\end{table*}

\begin{figure}
    \includegraphics[width=0.5\textwidth]{figures/rate_progression.pdf}
    \caption{{\bf Divergence and evolutionary rates of different Nextstrain clades over time.} Synonymous rates estimates are stable in time and fluctuate around the rate estimates for between clades. Non-synonymous rate estimates are highest for clades 19A to 20C that emerged early in the pandemic.
    \label{fig:rate_progression} }
\end{figure}

% The apparently higher non-synonymous rate suggests that at the beginning a large fraction of amino acid changes were beneficial to viral replication, not just the D614G mutation in the spike protein \citep{korber_tracking_2020}.
% This applies both to the changes within clades that were ultimately displaced by more successful variants, and inter-variant changes where the fraction of adaptive changes is even higher.



\section*{Purifying selection and mutation tolerance}

As described above, the rate of synonymous mutations is comparable within and between clades without a strong indication that this rate might have changed over time.
This is expected, as synonymous positions are rarely a locus of adaptation and tend to have a small effects on fitness in large parts of the genomes of RNA viruses \citep{zanini_vivo_2017} (outside of specific regions with important RNA elements or splice sites).
To quantify how much of the SARS-CoV-2 genome is constrained and how strongly purifying selection operates on different genomic regions, I made use of the ``rare mutations'' annotation provided by Nextclade.
Nextclade attaches each sequence to a reference tree and determines by which mutations it differs from the attachment point.
For each Pango lineage (as determined by Nextclade) \citep{rambaut_dynamic_2020,aksamentov_nextclade_2021}, I count how often these ``rare mutations'' (including reversions to the reference) are observed.
This way, for each position in the genome, one obtains the fraction of lineages with minor variation (excluding singletons).
I normalize this fraction against the relative rate of mutation away from the ancestral nucleotide (see Supp.~Fig.~\ref{fig:mutation_distribution}), and use this as a semi-quantitative proxy of mutational tolerance.

Simply splitting the genome into at 1st, 2nd, and 3rd positions of codons already reveals strong signatures of purifying selection, see Fig.~\ref{fig:fitness_costs}.
Between 15 and 20\% of first and second positions in codons show almost no variation, while half of these sites are less variable than the most constrained 10\% of third positions.
The median of variation at 3rd positions is more than double that at 1st and 2nd positions.

When split by open reading frame (see Supp.~Fig.~\ref{fig:fitness_costs_supp} and Supp.~Fig.~\ref{fig:fitness_landscape}), the most constrained regions are \emph{ORF1b} and \emph{M}, while \emph{ORF3a, ORF6, ORF7a, ORF7b}, and \emph{ORF8} show little evidence of constraint, consistent with frequently observed premature stop-mutations in some of these genes.
\emph{N} shows an intermediate pattern, possibly reflecting its mix of structured and unstructured regions.
Variation at 3rd positions is common and comparable between genes.
Only \emph{E} shows slightly less variation at 3rd positions than other genes with a notable dip in the middle of the gene around codon 35 (see Fig.~\ref{fig:fitness_landscape}).
Systematic difference in purifying selection on the amino acid sequence of different open reading frames have also been reported by \citet{rochman_ongoing_2021}.



\begin{figure}
    \includegraphics[width=0.5\textwidth]{figures/fitness_cost.pdf}
    \caption{{\bf Constraints on SARS-CoV-2 mutations.}
    Almost all 3rd codon positions tolerate mutations, while 1st and 2nd positions are strongly constrained.
    About half of the 1st and 2nd codon positions are less variable than the most constrained 10\% of 3rd positions (gray line).
    \label{fig:fitness_costs}}
 \end{figure}

 \begin{figure*}[tb]
    \includegraphics[width=\textwidth]{figures/fitness_landscape.pdf}
    \caption{{\bf Landscape of selective constraint along the SARS-CoV-2 genome.}
    Solid lines show sliding window smoothing of estimated mutational tolerance at 1st, 2nd, and 3rd positions for a window size of 20 and 7 (faint lines) sites.
    The markers at the bottom of the panels show qualitatively how many Pango lineages are differ at this position from the Wuhan-Hu-1 reference sequence.
    \label{fig:fitness_landscape}}
 \end{figure*}

\section*{Discussion}
The inferred evolutionary rate of RNA viruses often decreases with the time scale across which it is estimated \citep{wertheim_purifying_2011,ghafari_prisoner_2021}.
This effect can be particularly pronounced at the beginning of an outbreak following a host switch and has been attributed to incomplete purifying selection or methodological issues leading to inflated measures of diversity \citep{meyer_time_2015,ghafari_purifying_2022}.
In addition to segregating deleterious mutations, early viral evolution after a host switch can also be driven by anomalously fast adaptation.
A dramatic change in environment, e.g.~a host switch, likely results in many opportunities for mutations that increase fitness.
Such transient increases in the rate of adaptation are common in experimental evolution \citep{elena_virus_2007}.

Here, I showed that the deceleration during the first six months of the pandemic is only observed for non-synonymous mutations.
Since I analyzed evolution within short-lived clades of SARS-CoV-2 over a span of up to six months, purifying selection has comparable efficiency and should affect our estimates equally for all clades.
Similarly, sequencing artifacts are expected to increase divergence at all time points and affect synonymous and non-synonymous rate in similar ways.
The same applies to potential changes in the fidelity of the polymerase.
Nevertheless, the estimated non-synonymous evolutionary rate of clades circulating in late 2019 and early 2020 is about twice as high as that of clades in 2021 and 2022, while the synonymous rate does not change (see Fig.~\ref{fig:rate_summary} and Table \ref{tab:rates}).
One possible explanation is that the early evolutionary rate of SARS-CoV-2 was inflated by adaptive evolution.

Most of this apparent initial acceleration of within-variant evolution ceased in mid-2020 and even early variants like 20E accumulated non-synonymous changes at a rate of about 6 instead of 12 changes per year.
By that time, the number of non-synonymous differences relative to the root of the tree was small (5 in the case of 20E) and it is implausible that this small change would have exhausted the pool of beneficial mutations.
So why would the rate of adaptive evolution slow down?
Maybe it is just chance, after all it is only clades 19A and 19B where the rate is substantially higher.
Another possible explanation could be \emph{diminishing returns} epistasis.
Viruses with the S:D614G mutation showed faster replication kinetics compared to earlier variants \citep{korber_tracking_2020}.
The virus might therefore be operating closer to the maximal capacity at which cells can produce virions, reducing the scope for further optimization.
Other mutations then have smaller benefits, rise in frequency more slowly, and the effect of adaptation does not manifest itself over the life time of the clades studied here.
Such diminishing returns epistasis has been observed in experimental evolution with yeast \citep{kryazhimskiy_global_2014}.
Global epistasis of this nature can exist along side epistasis of specific amino acid changes as for example suggested for the \texttt{S:N501Y} mutation \citep{martin_emergence_2021,rochman_epistasis_2022,martin_selection_2022}.
Differences in diversification of SARS-CoV-2 with and without the S:D614G mutation where also observed in evolution experiments by \citet{amicone_mutation_2022}, who concluded that these differences were not due to a change in base line mutation rate but had selective origin.

An evolutionary rate of 6 synonymous changes per year at around 9700 positions corresponds to a per-site evolutionary rate of $6.2\times 10^{-4}$/site/year or $1.7\times 10^{-6}$/day, slightly higher than the estimated base-line mutation rate of $1.3\times 10^{-6}$/day \citep{amicone_mutation_2022}.
The total evolutionary rate within variants after mid 2020 (starting with 20E) is in the range of 9-16 changes per year, corresponding to a per site rate of $3 - 5\times 10^{-4}$/site/year, consistent with recent estimates by \citet{hill_origins_2022} and \citet{tay_emergence_2022}.

When considering only the putative founder genotype and date of origin of each variant, all variants so far are compatible with a back-bone evolutionary rate of 32 changes per year, corresponding to an per site rate of around $10^{-3}$/site/year (see Fig.~\ref{fig:rate_summary}).
This rate is a composite of episode of cryptic accelerated evolution (possibly in chronically infected individuals), and regular transmission chains of acute infections.
This estimate for the average rate is thus not inconsistent with results by \citet{hill_origins_2022} and \citet{tay_emergence_2022}, who estimated even higher rates specifically for the branches that gave rise to variants of concern.
Different clades and variants probably emerged in different ways under different circumstances.
Nevertheless, all clades and variants are compatible with a single ``back-bone'' molecular clock that runs substantially faster than the ``within variant clock''.
This accelerated back-bone clock is likely driven by exponential amplification of beneficial mutations.

In addition to neutral and potentially adaptive mutations, I also quantified purifying selection on the SARS-CoV-2 genome.
By analyzing the rate of mutations that spread only on short times scales within fine-grained Pango lineages, I estimated the level of constraint on different parts of the SARS-CoV-genome.
The great majority of 3rd positions in codons -- at which most mutations are synonymous -- do not show strong signatures of conservation.
The major open reading frames \emph{ORF1ab, S, N, E} and \emph{M} show clear signatures of purifying selection with around 50\% of sites being more constrained than the most constrained 10\% of 3rd positions.
Purifying selection analyzed here is operating over the time during which rare lineage circulate, which is typically a few weeks.
On shorter time scales, more strongly deleterious mutations will be observed.
On longer time scales, weaker deleterious effects can be quantified.
The degree of purifying selection is consistent with a similar analysis in HIV-1 that also inferred small fitness costs for most synonymous mutations, while half of the non-synonymous mutations are so deleterious that they are not even observed on short timescales \citep{zanini_vivo_2017}.

The SARS-CoV-2 genes \emph{ORF3a, ORF6, ORF7a, ORF7b} and \emph{ORF8} show little global signal of constraint and 1st and 2nd positions are as variable as 3rd positions (see Fig.~\ref{fig:fitness_costs}).
This lack of strong purifying selection in these ORFs does not necessarily imply that they don't matter or are not possible loci of adaptation -- but the exact amino acid sequence does not seem essential for their function.
Only a few regions show a clear signal of conservation at 3rd positions, notably a central region of \emph{E} and the ribosome slippage region at the beginning of \emph{ORF1b}.

The heterogeneity in evolutionary rates and the combination of adaptive evolution, approximately neutral mutations, and purifying selection complicate the interpretation of phylodynamic analysis.
Phylodynamics typically assumes that the mutation process is independent of the spread and epidemiology and that different sites evolve independently.
These assumptions are (approximately) true for neutral mutations that occur along every lineage with the same rate.
Strongly deleterious mutations don't spread and are only observed on terminal branches, similar to sequencing errors.
Weakly deleterious mutations can spread, but lineages that carry them tend to die out.
Overall, the deleterious mutations result in increased diversity on short time scales compared to longer ones, which can lead to time-dependent effective evolutionary rates \citep{wertheim_purifying_2011}.

The effects of adaptive evolution are harder to control and account for.
Since the number of sites that allow beneficial mutations is small, adaptive evolution tends to be very stochastic -- it is not the typical events, but the rare and extreme events that determine the course of adaptive evolution.
Unlike neutral evolution, the rate of adaptive evolution depends on the population size, stochastic nature of the transmission process, the environment, and previous adaptation \citep{neher_genetic_2013}.
These factors make extrapolation -- whether it is reconstruction of past events or scenarios of future evolution -- uncertain.
Standard phylogenetic or phylodynamic methods will underestimate this uncertainty.

If the excess of non-synonymous changes on the pandemic scale compared to within clade evolution is due to adaptive evolution, it would imply an adaptive rate of about 20 changes per year (plus insertion/deletions not considered here).
This process is likely to continue even in the more and more diverse immunological landscape of the human population in which the viral population spread and adapts.
But it is unclear whether SARS-CoV-2 continues to evolve in a saltatory fashion with repeated emergence of highly mutated variants, or whether we will see a transition to a more gradual adaptive process.
We currently observe multiple lineages with several amino acid changes and moderate transmission advantages emerging in a more stepwise fashion within 21L, 22A, 22B (BA.2, BA.4, BA.5), which could indicated a shift to a more influenza-like evolution.

\section*{Materials and Methods}

\subsubsection*{Data}
This analysis is based on SARS-CoV-2 sequencing data shared by researchers around the world via GISAID \citep{shu_gisaid_2017} (EPISET\_ID \href{10.55876/gis8.220812zf}{EPI_SET_220812zf}).
The bulk of the analysis can be reproduced using SARS-CoV-2 genomes shared via INSDC.
These data are available from Nextstrain at \url{http://data.nextstrain.org/files/ncov/open/metadata.tsv.gz} and \url{http://data.nextstrain.org/files/ncov/open/nextclade.tsv.gz}.
However, for some clades and variants, coverage in INSDC is low.

\subsubsection*{Analysis}
The code used to run this analysis is publicly available at \url{https://github.com/neherlab/SC2_variant_rates}.
It is organized as a snakemake workflow \citep{koster_snakemakescalable_2012}.
I used Nextclade data \texttt{2022-07-26T12:00:00Z} with reference sequence MN908947 (Wuhan-Hu-1).
The codon analysis ignores \emph{ORF9b} which is fully contained in \emph{N} but in a different reading frame.
Tabular files with estimates of evolutionary rates, the mutation distribution, and fitness costs are included in the git repository.

The fitness costs of each positions are estimated from the Nextclade columns \texttt{unlabeledSubstitutions} and \texttt{reversionSubstitutions} for each sequence that has QC-status \texttt{good}.
For each Pango lineage, I count how often each specific mutation was observed.
I then calculate how many lineages show minor variation for each mutation.
To avoid loss of signal through sequencing errors, I drop lineages at which only a single mutation at this position is observed.
To turn this count into a measure of mutational tolerance, I normalize the number of lineages with minor variation by the mutation rate away from the ancestral base at this position within the lineage.

\section*{Acknowledgements}
I gratefully acknowledge the work by 1000s of scientists and healthcare professionals that collect the samples, generate, and share the sequence data on which this analysis is built.
I am also grateful to Jesse Bloom, Louis du Plessis, and Florence D\'ebarre for valuable feed-back on the manuscript and to Cornelius Roemer and Emma Hodcroft for stimulating discussions on many aspects of SARS-CoV-2 evolution.

\bibliography{bib}

\clearpage
\appendix
\input{appendix}
\clearpage
\onecolumngrid
\input{appendix_rtt}
% \include{appendix_counts}
\end{document}
